code/baseStation/base_parser3.py

```diff
@@ -77,21 +89,20 @@ def image_get(self):
                 self.image.write(bytes(img_line[:(self.img_rcvd_bytes-self.image_size)]))
                 self.image.close()
             else:
-                self.image.write(bytes("".join(img_line)))
+                self.image.write(bytes(img_line))
         self.img_lastline = img_line
 
-
     def thermal_init(self):
         """
         Initializes receiving of a thermal image.
         Triggers when STATUS with thermal data is received
         """
 
         print("RECEIVING THERMAL", self.line[:15])
-        self.thermal = open(self.thermal_path+''.join(self.line[2:7])+"_"+str(time())+".txt", "w")
+        decoded = bytes(self.line).decode("ascii")
+        self.thermal = open(self.thermal_path+decoded[2:7]+"_"+str(time())+".txt", "w")
```

Comment:
Just use format please, will ya?

Reply:
These remains of py2...
Also applied formatting elsewhere where needed